server/e2e_test.go

```diff
@@ -166,31 +170,9 @@ func (h *testHarness) NewPool(poolSize int, value, version uint64,
 		allClients = append(allClients, c)
 
 		c.Connect()
-		c.Register(value, version)
-		if i < poolSize-1 {
-			// new players are announced while the pool is not full
-			h.AssertBroadcastNewPlayer(c, allClients)
-			h.AssertPoolStates([]testPoolState{
-				{
-					value:   value,
-					version: version,
-					players: len(allClients),
-					isFull:  false,
-				},
-			}, false)
-		}
+		isFullPool := i == poolSize-1
+		c.Register(value, version, allClients, isFullPool)
```

Comment:
Many of the changes are for readability. This is one of the functional changes. I encapsulated all the assertions and waiting related to registration in the action itself so that when it returns, the server should be in a known state.